core/shared/src/main/scala/scalaz/effect/KleisliIO.scala

```diff
@@ -190,9 +192,13 @@ object KleisliIO {
   }
 
   private[effect] final class Pure[E, A, B](apply0: A => IO[E, B]) extends KleisliIO[E, A, B] {
+    override val run: A => IO[E, B] = a => apply(a)
```

Comment:
Instead of doing this, change this:

```scala
private[effect] final class Pure[E, A, B](apply0: A => IO[E, B]) extends KleisliIO[E, A, B] {
```

to this:

```scala
private[effect] final class Pure[E, A, B](val run: A => IO[E, B]) extends KleisliIO[E, A, B] {
```

This will be faster because you won't be wrapping a closure with another closure.